editor/src/document/document_file.rs

```diff
@@ -332,23 +335,30 @@ impl DocumentMessageHandler {
 		layer_data(&mut self.layer_data, path)
 	}
 
-	pub fn backup(&mut self) {
+	pub fn backup(&mut self, responses: &mut VecDeque<Message>) {
 		self.document_redo_history.clear();
 		let new_layer_data = self
 			.layer_data
 			.iter()
 			.filter_map(|(key, value)| (!self.graphene_document.layer(key).unwrap().overlay).then(|| (key.clone(), *value)))
 			.collect();
-		self.document_history.push((self.graphene_document.clone_without_overlays(), new_layer_data))
+		self.document_undo_history.push((self.graphene_document.clone_without_overlays(), new_layer_data));
+
+		// Push the GetOpenDocumentsList message to the bus in order to update the save status of the open documents
+		responses.push_back(DocumentsMessage::GetOpenDocumentsList.into());
 	}
 
-	pub fn rollback(&mut self) -> Result<(), EditorError> {
-		self.backup();
-		self.undo()
+	pub fn rollback(&mut self, responses: &mut VecDeque<Message>) -> Result<(), EditorError> {
+		self.backup(responses);
+		self.undo(responses)
+		// TODO: Consider if we should check if the document is saved
 	}
 
-	pub fn undo(&mut self) -> Result<(), EditorError> {
-		match self.document_history.pop() {
+	pub fn undo(&mut self, responses: &mut VecDeque<Message>) -> Result<(), EditorError> {
+		// Push the GetOpenDocumentsList message to the bus in order to update the save status of the open documents
+		responses.push_back(DocumentsMessage::GetOpenDocumentsList.into());
```

Comment:
It feels a bit weird for these function to unconditionally push an update to the message queue, I'm wondering if there are situations where you would want to use the history operations without a message queue. But this probably helps to avoid errors

Reply:
(but you could return the message as the Ok() part, then clippy is going to complain if you don't use the `result but you could still use the function without a message queue)`

Reply:
That's a good point. However, do we want to allow this function to be run without the message queue? Currently, we want to send the GetOpenDocumentsList message at all times, so it would lead to code duplication that if missed at a later time could create a hard-to-track down bug.

Reply:
Yeah its fine as it is we can change it should it become a problem